quickwit/quickwit-serve/src/rest.rs

```diff
@@ -158,9 +158,13 @@ pub(crate) async fn start_rest_server(
         rest_listen_addr=?rest_listen_addr,
         "Starting REST server listening on {rest_listen_addr}."
     );
-    let serve_fut = hyper::Server::bind(&rest_listen_addr)
-        .serve(Shared::new(service))
-        .with_graceful_shutdown(shutdown_signal);
+
+    let serve_fut = async move {
```

Comment:
I'm okay with this. Note that dropping the server will not prevent current connections from working ([ref](https://github.com/hyperium/hyper/issues/2386#issuecomment-754912276))
I would either add a comment there or open a low-priority issue on the REST server graceful shutdown.

Reply:
ah good point. I am not sure why but it did fix the test terminating after the rest clients issue.